src/replays/replayitem.py

```diff
@@ -138,18 +150,26 @@ def update(self, message, client):
             else:
                 self.duration = time.strftime('%H:%M:%S', time.gmtime(seconds)) + "<br />&nbsp;playing"
         else:
-            self.duration = time.strftime('%H:%M:%S', time.gmtime(message["duration"]))
-        self.startHour = time.strftime("%H:%M", time.localtime(message['start']))
-        self.startDate = time.strftime("%Y-%m-%d", time.localtime(message['start']))
-        self.mod       = message["mod"]
+            endDt   = datetime.strptime(replay["attributes"]["endTime"], '%Y-%m-%dT%H:%M:%SZ') 
+            endDt   = endDt.replace(tzinfo=timezone.utc).astimezone(tz=None) #local time
+            self.duration = time.strftime('%H:%M:%S', time.gmtime((endDt - startDt).total_seconds()))
+            
+        self.startHour = startDt.strftime("%H:%M")
+        self.startDate = startDt.strftime("%Y-%m-%d")
+        
+        self.modid     = replay["relationships"]["featuredMod"]["data"]["id"]
+        self.mod       = message["featuredMods"][self.modid]["technicalName"]
 
         # Map preview code
         self.mapdisplayname = maps.getDisplayName(self.mapname)
       
         self.icon = maps.preview(self.mapname)
         if not self.icon:
-            self.client.map_downloader.download_preview(self.mapname, self._map_dl_request)
-            self.icon = util.THEME.icon("games/unknown_map.png")
+            if self.mapname == "unknown":
+                self.icon = util.THEME.icon("games/unknown_map.png")
+            else:    
+                self.client.map_downloader.download_preview(self.mapname, self._map_dl_request)
+                self.icon = util.THEME.icon("games/unknown_map.png")
```

Comment:
Unknown? `O_o`

Reply:
Strange condition: if mapname is unknown, then unknown, otherwise load preview and set icon to unknown...
Maybe icon should be unknown by default and don't touch it, if you don't know the map name